inst/rmd/h/anchor-sections/anchor-sections.js

```diff
@@ -0,0 +1,29 @@
+// Anchor sections v1.0 written by Atsushi Yasumoto on Oct 3rd, 2020.
+document.addEventListener('DOMContentLoaded', function() {
+  const h = document.querySelectorAll('h1, h2, h3, h4, h5, h6');
+
+  // Do nothing if sections are already anchored
+  if (Array.from(h).some(x => x.classList.contains('hasAnchor'))) {
+    return null;
+  }
+
+  // Use section id when pandoc runs with --section-divs
+  const section_id = function(x) {
+    return ((x.classList.contains('section') || (x.tagName === 'SECTION'))
+            ? x.id : '');
+  };
+
+  // Add anchors
+  h.forEach(function(x) {
+    const id = x.id || section_id(x.parentElement);
+    if (id === '') {
+      return null;
+    }
+    let anchor = document.createElement('a');
+    anchor.href = '#' + id;
+    anchor.classList = ['anchor-section'];
+    anchor.textContent = '#';
```

Comment:
One last comment: so we are hardcoding the symbol `#` here, and I think it can be more flexible to use the CSS `content` attribute of `a.anchor-section::before` to define the symbol. With the latter way, users will be able to override the symbol with CSS easily by themselves. Does that make sense?

Reply:
You are totally right! Improved via [`6c826f0` (#1884)](https://github.com/rstudio/rmarkdown/pull/1884/commits/6c826f0fb66b7c1c1027705690c9857ab79b258e)